src/qibolab/qubits.py

```diff
@@ -75,10 +76,25 @@ def __post_init__(self):
 
     @property
     def channels(self):
-        for channel in [self.readout, self.feedback, self.drive, self.flux, self.twpa]:
+        for name in self.CHANNEL_NAMES:
+            channel = getattr(self, name)
             if channel is not None:
                 yield channel
 
+    @property
+    def characterization(self):
+        """Dictionary containing characterization parameters."""
+        exclude_fields = self.CHANNEL_NAMES + ("name", "flux_coupler", "native_gates")
+        data = {fld.name: getattr(self, fld.name) for fld in fields(self) if fld.name not in exclude_fields}
+        # fix for dumping complex numbers in yaml
+        data["mean_gnd_states"] = str(data["mean_gnd_states"])
+        data["mean_exc_states"] = str(data["mean_exc_states"])
```

Comment:
I have the feeling we should find a better way than this, but it might be the simplest (at least for the time being).

Reply:
This was fixed in #504 because the same issue (complex not being serializable) appeared in qibocal too. The fix was incomplete because the type hint in the Qubit object and the test runcards were not updated but I fixed this in 487d8e16eb5fa4124771f981816b6b3744f1bae4.

We should also remember to update the runcards in the other repository for this change.